internal/guest/runtime/hcsv2/uvm.go

```diff
@@ -318,7 +332,7 @@ func (h *Host) CreateContainer(ctx context.Context, id string, settings *prot.VM
 		}
 	}
 
-	h.containers[id] = c
+	atomic.StoreUint32(&c.status, containerCreated)
```

Comment:
I dont think this is a good approach. You need an atomic for any reads as well. 

Given the lack of usage of atomics in the codebase, I can see this being incorrectly done (as it is currently in this PR). I think coming up with a more robust approach would be better. I'm not a Go expert so I'm not sure of the best pattern, but making all access atomic (store and load) without relying on authors of later changes to get it correct would be good.

Reply:
I'm afraid go-winio route will take too long. I'll just add getter/setter.